### PR TITLE
Add option to introduce manual pin

### PR DESCRIPTION
**Pull request by aguycalled**
Replaces #78 

TODO:

- Fix showing the pin code view

### CODE REVIEW

**Comment by aguycalled** [comment]
This PR is ready function-wise. Only the UI is broken.

image

Need to find a way to show the PinView and hide the current screen without unmounting the Stack Navigator.

Any idea? @tsejerome

File: assets/AssetIconsPack.iosasdas.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
```
Comment (tsejerome) [inline assets/AssetIconsPack.iosasdas.tsx:1]: can I know what is the use of this file?

File: contexts/SecurityProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  const [setManualPin, setSetManualPin] = useState<any>(() => {});
+  const [askManualPin, setAskManualPin] = useState<any>(() => {});
+
+  const [supportedType, setSupportedType] =
+    useState<SecurityAuthenticationTypes>(SecurityAuthenticationTypes.NONE);
+
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  useEffect((): void => {
+    AsyncStorage.getItem('AuthenticationType').then(async val => {
```
Comment (tsejerome) [inline contexts/SecurityProvider.tsx:94]: when are we using `AsyncStorage.getItem` and when are we using `useAsyncStorage`? we shd stick with one of them

File: contexts/SecurityProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+      if (creds) {
+        await new Promise((res, rej) => {
+          LocalAuth((error: any) => {
+            if (!error) {
+              res(true);
+            } else {
+              rej('failed local auth');
+            }
+          });
+        });
```
Comment (tsejerome) [inline contexts/SecurityProvider.tsx:193]: can we use `const res = await LocalAuth();` directly?

File: contexts/SecurityProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+          val == SecurityAuthenticationTypes.LOCALAUTH
+        ) {
+          // LocalAuth not available anymore
+          setError(
+            'Your previous authentication method is not available anymore. Please add a PIN code lock to your device or reinstall the app.',
+          );
+        } else {
+          setSupportedType(val as SecurityAuthenticationTypes);
+          setError(undefined);
+        }
```
Comment (tsejerome) [inline contexts/SecurityProvider.tsx:138]: I think it wud be better to add a little bit documentation for each if case such that we can easily understand when we come back to here. It was quite confused for me and took me a while to understand. (maybe i dont have enought context prevly)

**Review by tsejerome — CHANGES_REQUESTED** [review]

File: contexts/SecurityProvider.tsx (hunk 2 above)
Comment (aguycalled) [reply contexts/SecurityProvider.tsx:94]: yep, I will move to use `useAsyncStorage`

File: contexts/SecurityProvider.tsx (hunk 4 above)
Comment (aguycalled) [reply contexts/SecurityProvider.tsx:138]: I will add some docs on the file

File: contexts/SecurityProvider.tsx (hunk 3 above)
Comment (aguycalled) [reply contexts/SecurityProvider.tsx:193]: from the looks of it, I'd say LocalAuth does not return a promise but takes a callback as argument

**Comment by tsejerome** [comment]
LGTM

**Pull request merged**